src/js/components/Footer.js

```diff
@@ -1,61 +1,64 @@
 // (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP
 
-import React, { Component, PropTypes } from 'react';
-import pick from 'lodash/object/pick';
-import keys from 'lodash/object/keys';
+import React, { PropTypes } from 'react';
+import classnames from 'classnames';
 import Box from './Box';
 import SkipLinkAnchor from './SkipLinkAnchor';
 
-const CLASS_ROOT = "footer";
+const CLASS_ROOT = 'footer';
 
-export default class Footer extends Component {
-
-  render () {
-    var classes = [CLASS_ROOT];
-    var containerClasses = [CLASS_ROOT + "__container"];
-    var other = pick(this.props, keys(Box.propTypes));
-    if (this.props.size) {
-      classes.push(CLASS_ROOT + "--" + this.props.size);
-    } else if (this.props.large) { // Deprecated
-      classes.push(CLASS_ROOT + "--large");
-    }
-    if (this.props.className) {
-      classes.push(this.props.className);
+const Footer = props => {
+  if (!props.size) {
+    // Restore size value from deprecated props
+    if (props.small) {
+      props.size = 'small';
+    } else if (props.large) {
+      props.size = 'large';
     }
-    if (this.props.float) {
-      classes.push(CLASS_ROOT + "--float");
-      containerClasses.push(CLASS_ROOT + "__container--float");
+  }
+
+  let classes = classnames(
+    CLASS_ROOT,
+    props.className,
+    {
+      [`${CLASS_ROOT}--${props.size}`]: props.size,
+      [`${CLASS_ROOT}--float`]: props.float
     }
+  );
 
-    var footerSkipLink;
-    if (this.props.primary) {
-      footerSkipLink = (
-        <SkipLinkAnchor label="Footer" />
-      );
+  let containerClasses = classnames(
+    `${CLASS_ROOT}__container`,
+    {
+      [`${CLASS_ROOT}__container--float`]: props.float
     }
+  );
 
-    return (
-      <Box tag="footer" {...other} className={classes.join(' ')}
```

Comment:
Picking only the props Box supports or passing all of them does not change the result, since Box will only use what it needs. So it seemed to me like a little extra effort and for the sake of simplicity I removed the logic.

The only concern is overriding our own prop value with the one user passes. To prevent that, `{…props}` must take the first place while creating component instance. I did wrong here though. `tag` prop has to be placed after `{…props}`.